ibm/resource_ibm_private_dns_glb_pool.go

```diff
@@ -0,0 +1,418 @@
+		}
+		if region, ok := d.GetOk(pdnsGlbPoolRegion); ok {
+			updatePoolOptions.SetHealthcheckRegion(region.(string))
+		}
+		if _, ok := d.GetOk(pdnsGlbPoolSubnet); ok {
+			updatePoolOptions.SetHealthcheckSubnets(expandStringList(d.Get(pdnsGlbPoolSubnet).([]interface{})))
+		}
+		if _, ok := d.GetOk(pdnsGlbPoolOrigins); ok {
+			poolorigins := d.Get(pdnsGlbPoolOrigins).(*schema.Set)
+			updatePoolOptions.SetOrigins(expandPDNSGlbPoolOrigins(poolorigins))
+
+		}
+		_, detail, err := sess.UpdatePool(updatePoolOptions)
+		if err != nil {
+			return fmt.Errorf("Error updating pdns GLB Pool:%s\n%s", err, detail)
+		}
+	}
+
+	return resourceIBMPrivateDNSGLBPoolRead(d, meta)
+}
+
+func resourceIBMPrivateDNSGLBPoolDelete(d *schema.ResourceData, meta interface{}) error {
+	sess, err := meta.(ClientSession).PrivateDNSClientSession()
+	if err != nil {
+		return err
+	}
+
+	idset := strings.Split(d.Id(), "/")
+	DeletePoolOptions := sess.NewDeletePoolOptions(idset[0], idset[1])
+	response, err := sess.DeletePool(DeletePoolOptions)
+	if err != nil {
+		return fmt.Errorf("Error deleting pdns GLB Pool:%s\n%s", err, response)
+	}
+	_, err = waitForPDNSGlbPoolDelete(d, meta)
+	if err != nil {
+		return err
+	}
+	d.SetId("")
+	return nil
+}
+
+func resourceIBMPrivateDNSGLBPoolExists(d *schema.ResourceData, meta interface{}) (bool, error) {
+	sess, err := meta.(ClientSession).PrivateDNSClientSession()
+	if err != nil {
+		return false, err
+	}
+
+	idset := strings.Split(d.Id(), "/")
+
+	getPoolOptions := sess.NewGetPoolOptions(idset[0], idset[1])
+	response, detail, err := sess.GetPool(getPoolOptions)
+	if err != nil {
+		if response != nil && detail != nil && detail.StatusCode == 404 {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+func expandPDNSGlbPoolOrigins(originsList *schema.Set) (origins []dns.OriginInput) {
+	for _, iface := range originsList.List() {
+		orig := iface.(map[string]interface{})
+		origin := dns.OriginInput{
+			Name:        core.StringPtr(orig[pdnsGlbPoolOriginsName].(string)),
+			Address:     core.StringPtr(orig[pdnsGlbPoolOriginsAddress].(string)),
+			Enabled:     core.BoolPtr(orig[pdnsGlbPoolOriginsEnabled].(bool)),
+			Description: core.StringPtr(orig[pdnsGlbPoolOriginsDescription].(string)),
+		}
+		origins = append(origins, origin)
+	}
+	return
+}
+
+func waitForPDNSGlbPoolDelete(d *schema.ResourceData, meta interface{}) (interface{}, error) {
+	cisClient, err := meta.(ClientSession).PrivateDNSClientSession()
+	if err != nil {
+		return nil, err
+	}
+	idset := strings.Split(d.Id(), "/")
+	getPoolOptions := cisClient.NewGetPoolOptions(idset[0], idset[1])
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{pdnsGlbPoolDeletePending},
+		Target:  []string{pdnsGlbPoolDeleted},
+		Refresh: func() (interface{}, string, error) {
+			_, detail, err := cisClient.GetPool(getPoolOptions)
+			if err != nil {
+				if detail != nil && detail.StatusCode == 404 {
+					return detail, clusterDeleted, nil
+				}
```

Comment:
why are we returing clusterDeleted it shd be pdnsGlbPoolDeleted

Reply:
Handled